io.sloeber.core/src/io/sloeber/core/api/BoardDescriptor.java

```diff
@@ -918,6 +919,25 @@ public String getUploadCommand(ICConfigurationDescription confdesc) {
 		}
 		String ret = Common.getBuildEnvironmentVariable(confdesc,
 				"A.TOOLS." + upLoadTool.toUpperCase() + "." + action + ".PATTERN", "");
+		
+		// Below we handle the case where the port might have moved. This should be reasonably safe,
+		// since the port value that we have has to have been set, and it has to match a value from
+		// the port list ignoring any numbers on the end. If this specific scenario doesn't occur,
+		// then it will have no effect.
+		String oldComPort = getUploadPort();
+		if (!"".equals(oldComPort)) {
```

Comment:
As discussed in Slack I feel there should be a test for a key here. Do not uses space as that is a key for something else.